storage/include/storage/storage_metrics.h

```diff
@@ -21,32 +21,31 @@ namespace storage {
  * Recall that the in memory db is quite simple and therefor it has only few relevant metrics to collect.
  */
 class InMemoryStorageMetrics {
-  std::chrono::seconds metrics_update_interval_ = std::chrono::seconds(5);  // TODO: move to configuration
-  std::chrono::steady_clock::time_point last_metrics_update_ = std::chrono::steady_clock::now();
+  std::atomic<uint32_t> last_metrics_update_ = 0;
+  uint32_t metrics_update_interval_ = 100;  // We update the metrics once in 100 operations.
```

Comment:
We were actually updating the metrics every 5 seconds before. If we have lots of operations, doing it every 100 operations will update much more often. I am not sure if that is problematic or not, but just to note it. Maybe we can update less often?

Reply:
I think that if we have many operations we want to see the differences, so it makes sense to update on an operation based rather on time-based, WDY?
I can increase it ofcourse

Reply:
Yes, I agree with you. I am just unsure of how to define the operation count.

Reply:
@andrewjstone, any idea?

Reply:
We use 100ms updates in the replica thread. I would increase this to at least 1000. It really depends on throughput how frequently you want to do it unfortunately. You are going to have sync overhead too much if updating too quickly, or you are not going to see the stats you want if updating too slowly. This is why time based works better. 

I think the problem you ran into was that you are doing the check and update from multiple threads. You don't have to do that though. You can just have a timer in a separate thread that runs an update callback. This costs a thread for a timer, but most of the time it's just sleeping waiting on TimerFd in linux. Probably doable with std::async.

Reply:
done.
I preferred to create my own thread instead of using std::async because I don't want to spawn a new thread on every invocation.

Reply:
Agreed, having a separate thread makes it simpler and allows us to set periods. Thanks!